rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/terminal/TerminalIDEClient.java

```diff
@@ -131,7 +132,11 @@ public ISourceLocation resolveProjectLocation(ISourceLocation input) {
             return server.resolveProjectLocation(new SourceLocationParameter(input))
                 .get()
                 .getLocation();
-        } catch (InterruptedException | ExecutionException e) {
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
I assume this only restores the isInterruped state and not throws an exception itself?

Reply:
no, it actually interrupts the thread, which only happens at shutdown.

Not handling the interrupted exception this way can cause shutdown problems, where the JVM keeps running as the interrupts are swallowed.